packages/block-library/src/post-date/edit.js

```diff
@@ -118,10 +120,15 @@ export default function PostDateEdit( {
 									<PublishDateTimePicker
 										currentDate={ date }
 										onChange={ setDate }
-										is12Hour={ is12HourFormat(
-											siteTimeFormat
-										) }
+										is12Hour={ is12Hour }
 										onClose={ onClose }
+										dateOrder={
+											is12Hour
+												? /* translators: Order of day, month and year when the 12-hour clock is enabled. Available formats are 'dmy', 'mdy', 'ymd'. */
+												  _x( 'mdy', 'date order' )
+												: /* translators: Order of day, month and year when the 12-hour clock is disabled. Available formats are 'dmy', 'mdy', 'ymd'. */
+												  _x( 'dmy', 'date order' )
+										}
```

Comment:
```suggestion
										dateOrder={
											/* translators: Order of day, month and year. Available formats are 'dmy', 'mdy', 'ymd'. */
											_x( 'dmy', 'date order' )
										}
```

This is to maintain the current behavior, but if we don't have locales where the sorting changes depending on whether 12-hour clock is in effect, we might be able to change it to the above.